#### About

> Normal Distribution and Z-scores

- Normal Distribution, also known as the Gaussian distribution or bell curve, is a probability distribution that is widely used in statistics to model real-world phenomena such as height, weight, and test scores. The distribution has a symmetric, bell-shaped curve and is characterized by two parameters: the mean and the standard deviation.


- Z-score, also known as standard score, is a measure of how many standard deviations an observation or data point is away from the mean of the population. Z-scores are often used in statistical analysis to compare individual observations or data points across different datasets.



In [1]:
import numpy as np
from scipy.stats import norm

data = np.random.normal(loc=50, scale=10, size=100)

z_score = (60 - np.mean(data)) / np.std(data)

prob = 1 - norm.cdf(z_score)

print("Z-score: {:.2f}".format(z_score))
print("Probability of observing a value of 60 or higher: {:.2f}".format(prob))

Z-score: 0.98
Probability of observing a value of 60 or higher: 0.16


The Z score is 0.98 representing that the data is one std deviation above the mean of the population

Use cases of Z score include finding the creditworthiness of an org.